core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1070,29 +1075,26 @@ public static function generateMargin($arrValues, $strType='margin')
 	 */
 	public static function addToUrl($strRequest, $blnAddRef=true, $arrUnset=array())
 	{
-		$query = new Query(Environment::get('queryString'));
+		if (static::$objQuery === null)
+		{
+			static::$objQuery = Query::createFromRFC3986(Environment::get('queryString'));
```

Comment:
Why is this not from `Request` like the `_contao_referer_id` as well?

Reply:
Because the `_contao_referer_id` part is new and the rest is legacy code? I am fine with changing this, though.

Maybe we can get rid of `leagure/uri-components` completely if we use a combination of Symfony's `HeaderUtils::parseQuery()`, some custom logic and `http_build_query()` (like Symfony does in `Request::normalizeQueryString()`)?

Reply:
See 6d1f50dd22e02bc20f038b4a56b639b7352f146d